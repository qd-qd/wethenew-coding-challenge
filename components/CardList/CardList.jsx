import PropTypes from 'prop-types';

import Card from '@components/Card';
import styles from './CardList.module.css';

const CardList = ({ sneakers }) => (
  <section className={styles.container} data-testid="card-list">
    {sneakers.map(sneaker => (
      <Card
        key={sneaker.id}
        slug={sneaker.slug}
        price={sneaker.price}
        name={sneaker.name}
        brand={sneaker.brand}
        image={sneaker.shoes_pictures?.[0]}
      />
    ))}
  </section>
);

CardList.propTypes = {
  sneakers: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      brand: PropTypes.string,
      slug: PropTypes.string,
      release_date: PropTypes.string,
      price: PropTypes.string,
      size: PropTypes.arrayOf(PropTypes.string),
      shoes_pictures: PropTypes.arrayOf(PropTypes.string),
      description: PropTypes.string,
    }),
  ).isRequired,
};

export default CardList;
