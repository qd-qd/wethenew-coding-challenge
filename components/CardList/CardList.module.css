.container {
  display: grid;
  grid-gap: 1rem;
}

@media (min-width: 23.45em) {
  .container {
    grid-template-columns: repeat(auto-fit, 350px);
    justify-content: space-around;
  }
}
