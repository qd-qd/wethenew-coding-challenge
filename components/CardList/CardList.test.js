import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import sneaker from '@test/fixtures/sneaker.json';
import CardList from './CardList';

const sneakerList = new Array(3)
  .fill(sneaker)
  .map((value, index) => ({ ...value, id: `${index}` }));

describe('CardList', () => {
  test('displays greeting', () => {
    render(<CardList sneakers={sneakerList} />);

    expect(screen.getByTestId('card-list')).toBeDefined();
    expect(screen.getByTestId('card-list').childElementCount).toBe(
      sneakerList.length,
    );
  });

  test("doesn't break snapshot", () => {
    const { container } = render(<CardList sneakers={sneakerList} />);

    expect(container).toMatchSnapshot();
  });
});
