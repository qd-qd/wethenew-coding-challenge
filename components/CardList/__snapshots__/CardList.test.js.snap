// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList doesn't break snapshot 1`] = `
<div>
  <section
    class="container"
    data-testid="card-list"
  >
    <a
      data-testid="card-link"
      href="/fake-url"
    >
      <div
        class="container"
      >
        <img
          alt="The Fake shoes sneaker by Jest dude"
          class="image"
          src="/sneakers-placeholder.png"
        />
        <div
          class="section-infos"
        >
          <p
            class="name"
          >
            Fake shoes
          </p>
          <p
            class="price"
          >
            999
            <span>
              €
            </span>
          </p>
        </div>
        <p
          class="brand"
        >
          Jest dude
        </p>
      </div>
    </a>
    <a
      data-testid="card-link"
      href="/fake-url"
    >
      <div
        class="container"
      >
        <img
          alt="The Fake shoes sneaker by Jest dude"
          class="image"
          src="/sneakers-placeholder.png"
        />
        <div
          class="section-infos"
        >
          <p
            class="name"
          >
            Fake shoes
          </p>
          <p
            class="price"
          >
            999
            <span>
              €
            </span>
          </p>
        </div>
        <p
          class="brand"
        >
          Jest dude
        </p>
      </div>
    </a>
    <a
      data-testid="card-link"
      href="/fake-url"
    >
      <div
        class="container"
      >
        <img
          alt="The Fake shoes sneaker by Jest dude"
          class="image"
          src="/sneakers-placeholder.png"
        />
        <div
          class="section-infos"
        >
          <p
            class="name"
          >
            Fake shoes
          </p>
          <p
            class="price"
          >
            999
            <span>
              €
            </span>
          </p>
        </div>
        <p
          class="brand"
        >
          Jest dude
        </p>
      </div>
    </a>
  </section>
</div>
`;
